src/merkle/smt/mod.rs

```diff
@@ -62,6 +65,19 @@ pub(crate) trait SparseMerkleTree<const DEPTH: u8> {
     // PROVIDED METHODS
     // ---------------------------------------------------------------------------------------------
 
+    /// Creates a new sparse Merkle tree from an existing set of key-value pairs, in parallel.
+    #[cfg(feature = "concurrent")]
+    fn with_entries_par(entries: Vec<(Self::Key, Self::Value)>) -> Result<Self, MerkleError>
+    where
+        Self: Sized,
+    {
+        let (inner_nodes, leaves) = Self::build_subtrees(entries);
+        let leaves: BTreeMap<u64, Self::Leaf> =
+            leaves.into_iter().map(|(index, leaf)| (index.value(), leaf)).collect();
```

Comment:
Is there a reason why this step couldn't be combined with the loop on lines 479 - 486 below? Basically, we could have `build_subtrees()` return leaves as `BTreeMap<u64, Self::Leaf>` instead of `BTreeMap<LeafIndex<DEPTH>, Self::Leaf>`.